### PR TITLE
Update rand to 0.8

### PR DESCRIPTION
**Pull request by dancespiele**
Hello @mgeisler:
* I updated rand to 0.8 
* I updated rand_chacha = "0.3"
* I update the code according with the new version of rand
* I fix all the issues which rust clippy complain

### CODE REVIEW

**Comment by mgeisler** [comment]
Hi @dancespiele, wow, thanks very much for updating this!

I see you've put in a few other cleanups, could you please move them to other PRs? That will make it easier to see which changes are related to the rand 0.8 update and which are other (good) cleanups!

File: src/lib.rs [hunk 1]
```diff
@@ -130,7 +130,7 @@ impl<'a, R: Rng> MarkovChain<'a, R> {
         }
         // Sync the keys with the current map.
         self.keys = self.map.keys().cloned().collect();
-        self.keys.sort();
+        self.keys.sort_unstable();
```
Comment (mgeisler) [inline src/lib.rs:133]: I believe this is unrelated to the update, so please submit this in a separate PR, thanks! Then you can explain there why this is a great change :smile:

File: src/lib.rs [hunk 2]
```diff
@@ -348,14 +348,14 @@ fn join_words<'a, I: Iterator<Item = &'a str>>(mut words: I) -> String {
 ///
 /// [Wikipedia]: https://en.wikipedia.org/wiki/Lorem_ipsum
 /// [`LIBER_PRIMUS`]: constant.LIBER_PRIMUS.html
-pub const LOREM_IPSUM: &'static str = include_str!("lorem-ipsum.txt");
```
Comment (mgeisler) [inline src/lib.rs:351]: Please also move this and the other lifetime changes to a separate PR. I guess they come from Clippy? I would be happy to clean this up, as long as they go in separately.

File: src/lib.rs [hunk 3]
```diff
@@ -242,7 +242,7 @@ impl<'a, R: Rng> MarkovChain<'a, R> {
             map: &self.map,
             rng: &mut self.rng,
             keys: &self.keys,
-            state: state,
```
Comment (mgeisler) [inline src/lib.rs:245]: Please put this back — I like the explicit field names since it matches the other lines.

File: src/lib.rs [hunk 4]
```diff
@@ -103,7 +103,7 @@ impl<'a, R: Rng> MarkovChain<'a, R> {
         MarkovChain {
             map: HashMap::new(),
             keys: Vec::new(),
-            rng: rng,
```
Comment (mgeisler) [inline src/lib.rs:106]: Please also put this line back.

**Review by mgeisler — CHANGES_REQUESTED** [review]

**Comment by dancespiele** [comment]
sorry for my late reply, I will try to change today

**Comment by dancespiele** [comment]
I revert clean up, the changes that I didn't revert is required for the new version of rand 0.8

**Comment by mgeisler** [comment]
> @mgeisler I reverted clean up, the change that I didn't revert is required for the new version of rand 0.8

This looks really great now, thank you very much! I'll merge this now and do a release soon.

**Review by mgeisler — APPROVED** [review]

File: src/lib.rs [hunk 5]
```diff
@@ -297,7 +297,7 @@ fn is_ascii_punctuation(c: char) -> bool {
 }
 
 /// Capitalize the first character in a string.
-fn capitalize(word: &str) -> String {
```
Comment (mgeisler) [inline src/lib.rs]: It would be cool if you would submit this change as its own PR.

File: src/lib.rs [hunk 6]
```diff
@@ -348,14 +348,14 @@ fn join_words<'a, I: Iterator<Item = &'a str>>(mut words: I) -> String {
 ///
 /// [Wikipedia]: https://en.wikipedia.org/wiki/Lorem_ipsum
 /// [`LIBER_PRIMUS`]: constant.LIBER_PRIMUS.html
-pub const LOREM_IPSUM: &str = include_str!("lorem-ipsum.txt");
```
Comment (mgeisler) [inline src/lib.rs]: The removal of `'static` would also make a great PR.

**Pull request merged**